src/main/java/frc/robot/RobotContainer.java

```diff
@@ -50,41 +54,146 @@ public RobotContainer() {
         configureBindings();
     }
 
-    private void configureBindings() {
-        drive.registerTelemetry(logger::telemeterize);
-        drive.setDefaultCommand(new DriveWithJoysticks(drive, leftJoystick, rightJoystick));
+    private void configureSubsystems() {
```

Comment:
What if we cleaned this all up once and for all by making each subsystem configured in its own method? For example:

initDrivetrain
initIntake
initShooter
initVision
Etc.

Inside each of these functions we could do the switch case on if this is a real/sim/replay. From the configureSubsystems we could call the function based on the feature flag. I think this would be way more readable and make it clear what subsystems depend on each other.

For example initShooter might depend on initIntake being run, so that could be handled directly with feature flags rather than adding null checks

Reply:
Could work. We'd have to type the switch case a ton of times but honestly I can just copy paste it for every subsystem and then not worry about it again.

Reply:
We could add another layer to this and have configureSimulatedSubsystems / configureRealSubsystems / replay

That would avoid duplicate code but then we'd need to make a bunch of small init functions. If these init functions return a subsystem object we could put them all in a separate file as well to keep things super clean

Reply:
@jkleiber do you prefer always using {} after an if statement or do you like having single expression if-statements for if-statements that do a single thing?
Ex:
```java
        if (FeatureFlags.runIntake) {
            initIntake();
        }
        /* VS */
        if (FeatureFlags.runScoring) initScoring();
```

Reply:
My preference is to always have curly braces no matter what

Reply:
Okay same, that's what I did last night when I pushed that last commit.